### ISSUE TITLE
[mesheryctl] Enhancement to integration testing

### ISSUE THREAD

**Issue by metonymic-smokey** [meshery/meshery#4038]
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
A couple of days back, the `master` branch build was failing on `system start` when a meshconfig location was not specified using a flag. The default location of the meshconfig used in such cases was incorrectly marked as "".
The integration tests did not detect this failure since currently, the meshconfig is set during the tests. 

#### Desired Behavior
The integration tests should detect such a failure. 

#### Acceptance Tests

If `system start` fails when the default meshconfig is being used without specifying, the tests should be able to detect it. 

---

#### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

**Comment by metonymic-smokey** [comment]
Adding @piyushsingariya to the thread.

**Comment by shikharvashistha** [comment]
Duplicate #3988

**Comment by piyushsingariya** [comment]
> Duplicate #3988

it's different!

**Comment by shikharvashistha** [comment]
> > Duplicate #3988
> 
> it's different!

Oh yes
Apologies for misinterpreting the issue by its heading !

**Comment by san-coding** [comment]
Hey, I am new to open source, can I work on this issue ?

**Comment by piyushsingariya** [comment]
Yeah, definitely. but this issue is gonna be a bit tough, so be ready!

**Comment by san-coding** [comment]
thanks @piyushsingariya , I will need some helps, can you help me out ?

**Comment by piyushsingariya** [comment]
Yup always there to help, however you could also ping these amazing folks @sayantan1413 @suhaniii @Revolyssup!

**Comment by san-coding** [comment]
What files will i need to change to fix the integration tests ? @piyushsingariya

**Comment by piyushsingariya** [comment]
this error is from system_test.go

**Comment by san-coding** [comment]
thanks @piyushsingariya , I understood the issue and looked through the file for an hour but I am unable to figure out what changes to make, can you please help me out

**Comment by shikharvashistha** [comment]
> thanks @piyushsingariya , I understood the issue and looked through the file for an hour but I am unable to figure out what changes to make, can you please help me out
> 
> is this where the issue is ?
> https://github.com/meshery/meshery/blob/32bbc1789b21c1c2128e4457ad4da7629c482290/mesheryctl/internal/cli/root/system/system_test.go#L38-L46
> 
> or is it here
> 
> https://github.com/meshery/meshery/blob/32bbc1789b21c1c2128e4457ad4da7629c482290/mesheryctl/internal/cli/root/system/system_test.go#L80-L88
> 
> ## Which is the location flag ?

I want to add something to it. The issue specifies **location was not specified using a flag rather marked as  "" that is nil**  so, in my humble opinion something needs to be introduced rather than changed. Am I correct @piyushsingariya ?

**Comment by piyushsingariya** [comment]
You can look into the code and figure out that we actually setup the context by ourselves while running the tests. weather the integration tests are there to check if meshery will be running smoothly in a complete new env, but we are mistakenly setting up context ourselves. we gotta remove that, and pass a "-y" flag which will eventually create a new context and test-out if meshery is starting or not.

**Comment by san-coding** [comment]
> You can look into the code and figure out that we actually setup the context by ourselves while running the tests. weather the integration tests are there to check if meshery will be running smoothly in a complete new env, but we are mistakenly setting up context ourselves. we gotta remove that, and pass a "-y" flag which will eventually create a new context and set it up and test-out if meshery is starting or not.

Is this where you're setting the context ? Where exactly should we pass the `-y` flag so that the new context is created
https://github.com/meshery/meshery/blob/32bbc1789b21c1c2128e4457ad4da7629c482290/mesheryctl/internal/cli/root/system/system_test.go#L19

**Comment by piyushsingariya** [comment]
We have remove completely the setting up the context and pass "-y" as a flag in the args with "start" command.

**Comment by san-coding** [comment]
> We have remove completely the setting up the context thing and pass "-y" as a flag in the args with "start" command.

Do you mean we have to remove this ?
https://github.com/meshery/meshery/blob/32bbc1789b21c1c2128e4457ad4da7629c482290/mesheryctl/internal/cli/root/system/system_test.go#L19

and add such a section of code

https://github.com/meshery/meshery/blob/32bbc1789b21c1c2128e4457ad4da7629c482290/mesheryctl/internal/cli/root/system/system_test.go#L40-L46

sorry this might sound trivial to you , but I am not entirely sure what to do, thanks @piyushsingariya

**Comment by piyushsingariya** [comment]
yes that's what it is, If things are hard for you the only thing you should do is just be patient and spend time around the code. :smile:

**Comment by shikharvashistha** [comment]
Hi @san-coding are you still working on this ?

**Comment by san-coding** [comment]
Hey, I was a bit busy with university stuff, willl try to work on this now
